hbase-client/src/main/java/org/apache/hadoop/hbase/filter/RegexStringComparator.java

```diff
@@ -255,11 +258,20 @@ static interface Engine {
    * This is the default engine.
    */
   static class JavaRegexEngine implements Engine {
+    private static final Configuration conf = HBaseConfiguration.create();
```

Comment:
Can we find a way to pass this Configuration in?

Reply:
@Apache9 
I think it is possible.
`RSRpcService` pass configuration to `ProtobufUtil`.
https://github.com/apache/hbase/blob/f27823e62ddea58e827f68bb75289dbe13a1bd37/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java#L2456

->
```java
Get clientGet = ProtobufUtil.toGet(get, regionServer.conf);
```

`ProtobufUtil#toComparator` pass configuration to `ByteArrayComparable` when calling parseFrom method.
https://github.com/apache/hbase/blob/f27823e62ddea58e827f68bb75289dbe13a1bd37/hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ProtobufUtil.java#L1559
->
```java
public static ByteArrayComparable toComparator(ComparatorProtos.Comparator proto, Configuration conf) {
  return (ByteArrayComparable)parseFrom.invoke(null, value, conf);
}
```

I'll try to make the change.

Reply:
To pass configurations to RegexStringComparator, a new class ConfigurationHolder was created. It holds the configuration as a member variable. When HMaster and HRegionserver are started, it registers itself as an observer to the ConfigurationManager.